src/include/miopen/graphapi/graphapi_operation.hpp

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/errors.hpp>
+#include <miopen/graphapi/graphapi_tensor.hpp>
+
+#include <algorithm>
+#include <tuple>
+#include <vector>
+
+namespace miopen {
+
+namespace graphapi {
+
+class Operation
```

Comment:
You should rename this class to something `Convolution` related as the implementation is limited to `Convolutions` it seems.

Reply:
Not limited. This is a base for other operations as well. For each of the inputs we should add a value of `Operation::ArgumentId` enum or reuse an existing ones, for example `ArgumentId::X` for any single input operation.

Reply:
Then we don't need `pair`

Reply:
X-Y-W sounds like convolution parameters.
Also the minimum number of parameters for operations is 2: 1 input and 1 output tensor.
But for the RNG it can be even 0 input tensors and 1 output (and up to 3 scalar parameters, but it depends, each of them can be an input tensor).

Are you going to enumerate all kind of operand ids here?